src/Sarif/AnalyzeContextBase.cs

```diff
@@ -221,6 +225,24 @@ public virtual void Dispose()
             var disposableLogger = this.Logger as IDisposable;
             disposableLogger?.Dispose();
             this.Logger = null;
+
+            if (TraceEventSession != null)
+            {
+                TraceEventSession?.Flush();
+
+                if (TraceEventSession?.EventsLost > 0)
```

Comment:
>TraceEventSession? [](http://example.com/codeflow?start=20&length=18)

Same point here: the null check is redundant

Reply:
Yes, thanks, will follow up on this in next change.

Reply:
Had to reformat code anyway, so worked in your notes, thanks for the feedback.